config/routes.rb

```diff
@@ -20,9 +20,13 @@
     resource :vote, controller: 'comments/votes', only: [:create, :destroy]
   end
 
-  get ':username' => 'users/lists#index', as: :user_profile
+  get ':reference_id/:id/delete' => 'links#destroy', as: :destroy_link
+  get ':username/:list_id/:id/update' => 'references#edit_abstract', as: :edit_abstract
```

Comment:
These are the routes that `resources :dogs` generates, which are RESTful in the Rails sense, and which we should try to emulate:

```
  Prefix Verb   URI Pattern              Controller#Action
    dogs GET    /dogs(.:format)          dogs#index
         POST   /dogs(.:format)          dogs#create
 new_dog GET    /dogs/new(.:format)      dogs#new
edit_dog GET    /dogs/:id/edit(.:format) dogs#edit
     dog GET    /dogs/:id(.:format)      dogs#show
         PATCH  /dogs/:id(.:format)      dogs#update
         PUT    /dogs/:id(.:format)      dogs#update
         DELETE /dogs/:id(.:format)      dogs#destroy
```